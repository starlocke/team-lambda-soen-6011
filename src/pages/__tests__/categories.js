import React from "react"
import renderer from "react-test-renderer"
import CategoriesPage from "../categories"
import { StaticQuery, Link} from "gatsby"
// import Layout from "../../components/layout/layout"
beforeEach(() => {
  const originalError = console.error;
  console.error = jest.fn();
  // jest.mock("../../components/layout/layout", () => () => ('hi'));
  // jest.mock("../../components/seo", () => () => ());
    StaticQuery.mockImplementationOnce(({ render }) =>
      render({
        site: {
          siteMetadata: {
            title: `my title`,
          },
        },
      })
    )

//     jest.mock(Layout, () => () => (<div>Hello World</div>));

// jest.mock("../../components/layout/layout");
})

describe("CategoriesPage", () => {
  it("renders correctly", () => {
    // jest.mock("../../components/layout/layout");
    // jest.mock("../../components/layout/layout", () => {
    //   return {
    //     __esModule: true,
    //     default: () => {
    //       return <div></div>;
    //     },
    //   };
    // });
    const data = {
        "site": {
          "siteMetadata": {
            "title": "my title"
          }
        },
        "allMarkdownRemark": {
          "group": [
            {
              "fieldValue": "tag 1",
              "totalCount": 1
            },
            {
              "fieldValue": "tag 2",
              "totalCount": 10
            }
          ]
        }
      }
    
      
    const tree = renderer.create(<CategoriesPage data={data} />).toJSON();
    expect(tree).toMatchSnapshot()
  })
})