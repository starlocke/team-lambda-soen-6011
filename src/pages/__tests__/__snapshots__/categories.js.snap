// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesPage renders correctly 1`] = `
Array [
  <header
    style={
      Object {
        "background": "rebeccapurple",
        "marginBottom": "1.45rem",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          my title
        </a>
      </h1>
    </div>
  </header>,
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0px 1.0875rem 1.45rem",
        "paddingTop": 0,
      }
    }
  >
    <div>
      <div
        className="search__container"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <label
          htmlFor="searchInput"
        >
          Search All Skill Pages
        </label>
        <input
          autoComplete="off"
          className="search__input"
          id="searchInput"
          onChange={[Function]}
          placeholder="Query"
          type="text"
          value=""
        />
        <div
          className=""
        >
          
        </div>
      </div>
      <noscript>
        <form
          action="https://www.google.com/search"
          method="GET"
        >
          <input
            name="q"
            type="hidden"
            value="site:sestopia-gatsby.herokuapp.com"
          />
          <label>
            Search Sestopia:
            <br />
            <input
              name="q"
              placeholder="Query"
              type="text"
              value=""
            />
          </label>
        </form>
      </noscript>
    </div>
    <main>
      <div>
        <h1>
          Categories
        </h1>
        <ul>
          <li>
            <a
              href="/categories/tag-1/"
            >
              tag 1
               (
              1
              )
            </a>
          </li>
          <li>
            <a
              href="/categories/tag-2/"
            >
              tag 2
               (
              10
              )
            </a>
          </li>
        </ul>
      </div>
      <a
        href="/"
      >
        ‚Üê Back to the Home Page
      </a>
      <footer
        className="sc-AxjAm ipahVA"
      >
        SESTOPIA - An Information Intensive Application
      </footer>
    </main>
  </div>,
]
`;
